docker/Dockerfile.onnx-mlir-dev

```diff
@@ -48,6 +48,10 @@ RUN LLVM_PROJECT_ROOT=${WORK_DIR}/llvm-project \
             -j${NPROC} \
             check-onnx-backend-numerical \
     && make check-docs \
+# Run the cppcheck scanner on s390x
+    && if [ "$(uname -m)" = "s390x" ]; then \
+           ${ONNX_MLIR_ROOT}/utils/cppcheck_scan/cppcheck-scan.py; \
+       fi \
     && rm -f Debug/bin/*Test Debug/bin/Perf* Debug/bin/Test* \
```

Comment:
cppcheck is a static check. It should be run as early as possible. No point in running all the builds and tests successfully only later failing the cppcheck. I would move it up and have a separate RUN for it, right after
```
# Copy onnx-mlir from the pull request
COPY . onnx-mlir
```

Reply:
The cppcheck is being invoked with the project file from cmake - `onnx-mlir/build/compile_commands.json`.  Moving the cppcheck up to after that file is created.